unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -2043,6 +2064,19 @@ public SimObjPhysics[] VisibleSimObjs(bool forceVisible)
             }
         }
 
+        protected SimObjPhysics[] GetAllVisibleSimObjPhysicsFilter(IEnumerable<SimObjPhysics> simObjs, Camera agentCamera, float maxDistance) 
+        {
+            List<SimObjPhysics> visible = new List<SimObjPhysics>();
+            foreach(var sop in simObjs)
+            {
+                if(isSimObjVisible(agentCamera, sop, this.maxVisibleDistance)) 
+                {
+                    visible.Add(sop);
+                }
+            }
+            return visible.ToArray();
+        }
+
```

Comment:
I still don't think this logic is equivalent to the old visibility logic. The main problem is with long objects, imagine the following setting:

1. We have a plunger laying on its side angled away from us so that half of the plunger is within the visibility distance and half is beyond the visibility distance.
2. The part of the plunger that is within the visibility distance is occluded while the portion of the plunger that is outside our visibility distance is not.

With this new logic, the plunger is considered **not visible** but, with the old logic, it **is visible**. This is because the OverlapCapsule call will add the plunger to the `colliders_in_view` list after which we only require that a single visibility point be visible at any distance for the object to be considered visible.

Reply:
This is true, but this new behavior may be more consistent as the default moving forward. It didn't make much sense in this example to have a plunger laying lengthwise still be visible in that way where the closer visibility points are obstructed and only the further away ones outside of the visibility distance are visible.

I think this is more of a question of which behavior seems more appropriate given the context of visual tasks that might require one behavior or another

Reply:
I agree that the new behavior is more "reasonable" in some sense but as it fundamentally breaks backwards compatibility this isn't something we can just release. Perhaps it can be saved for our 3.0.0 release.

Reply:
In generally I think we have to be better in THOR about sticking to the semantic versioning guidelines (https://semver.org/) as, currently, even patch updates frequently break backwards compatibility.

Reply:
Luca, yes you are correct.  I hadn't considered that case.  I feel the new implementation makes more sense and is much easier to explain to others/document - it basically boils down > 1 visibility point within the maxVisibleDistance.  Under the old scheme you could also have something like an extremely long table that is obscured well outside the visibilitydistance, but one point 3m away is visible.  I don't think it makes sense to consider that "visible" as that implies its interactable.

We have a couple options:
1) I can modify the code to match the logic of the existing code
2) We do nothing and accept the new change
3) Modify the existing code to match the current logic, but also add a parameter to support the new scheme so that we can experiment with results

I would prefer we do 3 since I prefer the new logic, but there are unknown consequences to changes the visibility logic and we should run models with/without the new logic.

Reply:
My vote is also for 3. I believe in the next big version update the default should be the new behavior with the option of the old, but for now making the new scheme optional while retaining the current behavior sounds good.

Reply:
I'm happy with 3 and totally agree that the new logic is more sensible. I would even be happy with deprecating the old code visibility code on the next major version release.